src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -86,140 +85,110 @@ bool vtkDataMeshReader::read(const QString& path) {
-            parseHeaderVtk(header, medData);
-        }
-        else if (extension == "vtp") // VTP files
+        try
         {
             dataSet->Read(path.toLocal8Bit().constData());
             medData->setData(dataSet);
 
-            // Extract field data from the xml file
-            vtkSmartPointer<vtkXMLPolyDataReader> reader = vtkSmartPointer<vtkXMLPolyDataReader>::New();
-            reader->SetFileName(path.toLocal8Bit().constData());
-            reader->Update();
-
-            if (reader->GetOutput()->GetFieldData())
+            if (!extractMetaData(path, dataSet))
             {
-                parseHeaderVtp(reader->GetOutput()->GetFieldData(), medData);
+                qDebug() << metaObject()->className() << ": no metadata found in " << path;
             }
+
+            setProgress(100);
+            return true;
         }
-        else if ((extension == "mesh") || (extension == "obj")) // MESH or OBJ files
+        catch (vtkErrorCode::ErrorIds error)
         {
-            try
-            {
-                dataSet->Read(path.toLocal8Bit().constData());
-            } catch (...)
-            {
-                qDebug() << "vtkDataMeshReader::read -> loading the vtkDataMesh failed";
-                return false;
-            }
+            qDebug() << metaObject()->className() << ": " << vtkErrorCode::GetStringFromErrorCode(error);
+            return false;
         }
-        else
+        catch (...)
         {
+            qDebug() << metaObject()->className() << ": error reading " << path;;
             return false;
         }
     }
-
-    setProgress(100);
-    return true;
+    else
+    {
+        return false;
+    }
 }
 
-QStringList vtkDataMeshReader::metaDataKeysToCopy()
+bool vtkDataMeshReader::extractMetaData(QString path, vtkMetaDataSet* dataSet)
 {
-    QStringList keys;
-
-    keys << medMetaDataKeys::PatientID.key()
-         << medMetaDataKeys::PatientName.key()
-         << medMetaDataKeys::Age.key()
-         << medMetaDataKeys::BirthDate.key()
-         << medMetaDataKeys::Gender.key()
-         << medMetaDataKeys::Description.key()
-         << medMetaDataKeys::StudyID.key()
-         << medMetaDataKeys::StudyDicomID.key()
-         << medMetaDataKeys::StudyDescription.key()
-         << medMetaDataKeys::Institution.key()
-         << medMetaDataKeys::Referee.key()
-         << medMetaDataKeys::StudyDate.key()
-         << medMetaDataKeys::StudyTime.key()
-         << medMetaDataKeys::Performer.key()
-         << medMetaDataKeys::Report.key()
-         << medMetaDataKeys::Protocol.key()
-         << medMetaDataKeys::Origin.key()
-         << medMetaDataKeys::AcquisitionDate.key()
-         << medMetaDataKeys::AcquisitionTime.key()
-         << medMetaDataKeys::Modality.key()
-         << medMetaDataKeys::Orientation.key();
-
-    return keys;
+    if (extractMetaDataFromFieldData(dataSet) || extractMetaDataFromHeader(path, dataSet) || extractCartoMetaData(dataSet))
```

Comment:
Can't we separate these cases?
I think it's troublesome if a dataset can go in more than one method: for instance export a mesh the classic way, then load it and export in CARTO (enter patient Name, ID, etc...); load it again. In the db, the patient name should be the one you typed while exporting to Carto, but it's not. It's he original patient name.
It might sound vicious to do all these exports/imports but for now we send our "customers" vtk meshes and they export them in CARTO format, so it's quite a common case.